enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/plugin/PluginRealmTest.java

```diff
@@ -0,0 +1,154 @@
+    private AssertableLogProvider log = new AssertableLogProvider();
+    private SecurityLog securityLog = new SecurityLog( log.getLog( this.getClass() ) );
+
+    @Test
+    public void shouldLogToSecurityLogFromAuthPlugin() throws Throwable
+    {
+        PluginRealm pluginRealm = new PluginRealm( new LoggingAuthPlugin(), config, securityLog, Clock.systemUTC(),
+                mock( SecureHasher.class ) );
+        pluginRealm.initialize( IGNORED );
+        assertLogged( "LoggingAuthPlugin" );
+    }
+
+    @Test
+    public void shouldLogToSecurityLogFromAuthenticationPlugin() throws Throwable
+    {
+        PluginRealm pluginRealm = new PluginRealm(
+                new LoggingAuthenticationPlugin(),
+                null,
+                config, securityLog, Clock.systemUTC(), mock( SecureHasher.class ) );
+        pluginRealm.initialize( IGNORED );
+        assertLogged( "LoggingAuthenticationPlugin" );
+    }
+
+    @Test
+    public void shouldLogToSecurityLogFromAuthorizationPlugin() throws Throwable
+    {
+        PluginRealm pluginRealm = new PluginRealm(
+                null,
+                new LoggingAuthorizationPlugin(),
+                config, securityLog, Clock.systemUTC(), mock( SecureHasher.class ) );
+        pluginRealm.initialize( IGNORED );
+        assertLogged( "LoggingAuthorizationPlugin" );
+    }
+
+    private void assertLogged( String name )
+    {
+        log.assertExactly(
+                inLog( this.getClass() ).info( format( "{plugin-%s} info line", name ) ),
+                inLog( this.getClass() ).warn( format( "{plugin-%s} warn line", name ) ),
+                inLog( this.getClass() ).error( format( "{plugin-%s} error line", name ) )
+            );
+    }
+
+    private class LoggingAuthPlugin extends LoggingAdapter implements AuthPlugin
+    {
+        @Override
+        public String name()
+        {
+            return "LoggingAuthPlugin";
+        }
+
+        @Override
+        public AuthInfo authenticateAndAuthorize( AuthToken authToken ) throws AuthenticationException
+        {
+            return null;
+        }
+    }
+
+    private class LoggingAuthenticationPlugin extends LoggingAdapter implements AuthenticationPlugin
+    {
+        @Override
+        public String name()
+        {
+            return "LoggingAuthenticationPlugin";
+        }
+
+        @Override
+        public AuthenticationInfo authenticate( AuthToken authToken ) throws AuthenticationException
+        {
+            return null;
+        }
+    }
+
+    private class LoggingAuthorizationPlugin extends LoggingAdapter implements AuthorizationPlugin
+    {
+        @Override
+        public String name()
+        {
+            return "LoggingAuthorizationPlugin";
+        }
+
+        @Override
+        public AuthorizationInfo authorize( Collection<PrincipalAndRealm> principals )
+        {
+            return null;
+        }
+    }
+
+    abstract class LoggingAdapter extends RealmLifecycle.Adapter
```

Comment:
Not requesting change in this PR, but perhaps it would be nice to have a somewhat sophisticated test `PluginRealm` available to use (even as an abstract class that tests could extend), which would contain implementations for these interfaces? I'm just envisioning that we will have (or have already) several test classes that could use such a test infrastructure instead of having to tailor new private classes. WDYT?